### PR TITLE
SQ-237/Convert Venue domain to Kotlin

### PR DESCRIPTION
**Pull request by rock3r**
Little more than an automatic conversion in most cases; some additional hand-optimisations in the page view class were required.

### CODE REVIEW

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import net.squanchy.home.Loadable
+import net.squanchy.imageloader.ImageLoader
+import net.squanchy.imageloader.ImageLoaderInjector
+import net.squanchy.navigation.Navigator
+import net.squanchy.support.ContextUnwrapper.unwrapToActivityContext
+import net.squanchy.venue.domain.view.Venue
+
+class VenueInfoPageView : CoordinatorLayout, Loadable {
+
+    constructor(context: Context, attrs: AttributeSet) : this(context, attrs, 0)
```
Comment (quiro91) [inline app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt]: We can probably remove this constructor and just annotate the other one with `@JvmOverloads`

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            Html.fromHtml(description)
+        }
+    }
+
+    private fun loadMap(imageView: ImageView, mapUrl: String, imageLoader: ImageLoader) {
+        imageLoader.load(mapUrl).into(imageView)
+    }
+
+    private fun updateMapClickListenerWith(venue: Venue) {
+        venueMap.setOnClickListener { v -> navigator.toMapsFor(venue) }
```
Comment (quiro91) [inline app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt]: No need for `v ->`

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                }
+                else -> false
+            }
+        }
+    }
+
+    override fun startLoading() {
+        subscription = service.venue()
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ this.updateWith(it) })
```
Comment (quiro91) [inline app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt]: Not really a big deal, but we can remove the parenthesis here:
`subscribe { this.updateWith(it) }`

**Review by quiro91 — APPROVED** [review]
moar Kotlin <3

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt (hunk 3 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt]: fa1dd163836f8265c8b3cbd87e81123967a44197

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt (hunk 2 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt]: fa1dd163836f8265c8b3cbd87e81123967a44197

File: app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/venue/VenueInfoPageView.kt]: 177667447d606c3fe4097de2e9b35b4df3d13ee0

File: app/src/main/java/net/squanchy/venue/VenueInfoInjector.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package net.squanchy.venue
+
+import android.support.v7.app.AppCompatActivity
+
+import net.squanchy.injection.ActivityContextModule
+import net.squanchy.injection.ApplicationInjector
+import net.squanchy.navigation.NavigationModule
+
+internal object VenueInfoInjector {
```
Comment (fourlastor) [inline app/src/main/java/net/squanchy/venue/VenueInfoInjector.kt:9]: you don't need this object, you can simply have a top level function :)

**Pull request merged**